### ISSUE TITLE
[FeTS client] json body is typed as "never" when using POST requests

### ISSUE THREAD

**Issue by romaindurand** [ardatan/feTS#549]
**Bug description**

When using the `post()` method provided by the feTS client on the `"/users/{user_id}/playlists"` route of the Spotify API, the `json` field is incorrectly typed as `never` even though the Spotify open API schema provides informations about the required json structure for such a request.

Screenshot that confirms the spotify OAS has all the necessary informations to strongly type the `json` field : 

![Capture d’écran de 2023-09-07 11-58-55](https://github.com/ardatan/feTS/assets/4068963/b5afa542-50e9-452f-a2df-7452edc14961)

Screenshots that shows the `json` field typed as `never` :

![Capture d’écran de 2023-09-07 11-31-55](https://github.com/ardatan/feTS/assets/4068963/45400fac-cf52-4af8-a5fd-1aac1162e8ec)
![Capture d’écran de 2023-09-07 11-23-42](https://github.com/ardatan/feTS/assets/4068963/2e2f60cd-f570-45b7-83c8-33d533c840f8)

From my understanding of `typing/client/types.d.ts`, this `json` field should be strongly typed from the OAS so I don't get the expected behaviour : 

![Capture d’écran de 2023-09-07 11-53-35](https://github.com/ardatan/feTS/assets/4068963/0fe32032-a609-463b-88b6-397f033e2725)

***

**Steps to reproduce the behavior :** 

Here's a reproduction repository (the branch that demonstrates the bug is `bug/post-json`) : https://github.com/romaindurand/fets-test/blob/bug/post-json/client.ts

***

**Expected behavior**

json field should be strongly typed according to the provided open api specification schema

***

**Environment:**

- OS: ubuntu 20
- NodeJS: 18.15.0

***

**This library is awesome and I really hope this can be fixed (or explained). Thank you !**

**Comment by nmss** [comment]
This can be fixed by replacing
```
type FixMissingAdditionalProperties<T> = T extends {
    type: 'object';
    properties: any;
} ? T & {
    additionalProperties: false;
} : T;
```
with
```
type FixMissingAdditionalProperties<T> = T extends {
    type: 'object';
    additionalProperties: never;
} ? T & {
    additionalProperties: false;
} : T;
```

or by replacing it with

type FixMissingAdditionalProperties<T> = T extends {
    type: 'object';
    properties: any;
} ? Omit<T, 'additionalProperties'> & {
    additionalProperties: false;
} : T;

But I'm not sure of the implication of those changes

**Comment by ardatan** [comment]
Would you create a PR for this change?

**Comment by romaindurand** [comment]
> Thanks for all your feedback <3 Would you create a PR for this change?

I'm working on it !

**Comment by ardatan** [comment]
Fixed by #550 ! Thanks for your feedback and the PR!
